framework/src/main/java/org/checkerframework/common/aliasing/AliasingVisitor.java

```diff
@@ -271,7 +265,37 @@ private boolean canBeLeaked(Tree exp) {
         AnnotatedTypeMirror type = atypeFactory.getAnnotatedType(exp);
         boolean isMethodInvocation = exp.getKind() == Kind.METHOD_INVOCATION;
         boolean isNewClass = exp.getKind() == Kind.NEW_CLASS;
-        return type.hasExplicitAnnotation(Unique.class) && !isMethodInvocation && !isNewClass;
+        boolean isUniqueClassFlag = isUniqueClass(exp);
+        if (isUniqueClassFlag) {
+            return type.hasAnnotation(Unique.class) && !isMethodInvocation && !isNewClass;
+        } else {
+            return type.hasExplicitAnnotation(Unique.class) && !isMethodInvocation && !isNewClass;
+        }
+    }
+
+    /**
+     * Returns true if class of tree expression {@code exp} has type {@code @Unique}
+     *
+     * @param exp the Tree to check
+     * @return boolean true if class if unique and false otherwise
+     */
+    private boolean isUniqueClass(Tree exp) {
+        AnnotatedTypeMirror type = atypeFactory.getAnnotatedType(exp);
+        Element el = types.asElement(type.getUnderlyingType());
+        if (el != null) {
```

Comment:
It might be easier as `if (el == null) { return false; }` and then the same for `annoMirrors`.

Reply:
I have separated out el and annoMirrors checks instead of the nested if conditions. If either of them is null, the method returns false